packages/context/src/binding-inspector.ts

```diff
@@ -85,16 +92,21 @@ export function asProvider<T>(
  * A factory function to create a template function to bind the target class
  * as a class or `Provider`.
  * @param target - Target class, which can be an implementation of `Provider`
+ * or `DynamicValueProviderClass`
  *
  * @typeParam T - Value type
  */
 export function asClassOrProvider<T>(
```

Comment:
the focus of this PR is the change in this method , asClassOrProvider , correct ? and that is for allowing @bind() decorator and createBindingFromClass() to handle dynamic value providers is it ?

Reply:
Yes.